### PR TITLE
Feature/spaceplot tool

### LINKED ISSUE #275: godas monitoring tools for experiment runs
To plot bkg, incr and ombg, we develop plotting tools to monitor experiments.

### PR DESCRIPTION
**Pull request by mjhossen**
## Description

This PR adds two bash scripts to plot observation space, background vector and increment available in the experiment folder.

**To plot background vector:**
```console
bash soca_plotfield.sh -x EMC2 -v bkg -s 20210601 -e 20210601 -p path/to/file -l fct_length
```
**To plot increment:**
```console
bash soca_plotfield.sh -x EMC2 -v incr -s 20210601 -e 20210601 -p path/to/file -l fct_length
```
To plot ombg:
```console
bash godas_plotobs.sh -x EMC3 -v obs_out -s 20210201 -e 20210202 -p path/to/file -l fct_length
```

Fixes #275

### CODE REVIEW

File: monitoring/Readme.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+To plot the fields one can use the python scripts: soca_plotfield.py
```
Comment (guillaumevernieres) [inline monitoring/Readme.md]: 1 - The convention is to name that file README.md
2 - `.md` stands for markdown. Make use of it!

File: monitoring/godas_plotobs.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
+#!/home/gvernier/anaconda3/bin/python
```
Comment (guillaumevernieres) [inline monitoring/godas_plotobs.py]: Does not belong in this PR ... I just merge #272 , update the develop branch in your sand box and merge it in to your branch.

File: monitoring/godas_plotobs.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+cycle_dir=/work/noaa/marine/jhossen/$exp
+echo $cycle_dir
+cdir=$PWD/$exp
+mkdir $cdir
+stv=obs_out
+
+date_YMDH=$(date -ud "$start_date")
+YMDH=$(date -ud "$date_YMDH " +%Y%m%d%H )
+
+#END_YMDH=$(date -ud "$date_YMDH + ${num_days} days " +%Y%m%d%H )
```
Comment (guillaumevernieres) [inline monitoring/godas_plotobs.sh]: Remove comment

File: monitoring/godas_plotobs.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        python ./godas_plotobs.py -f $datadir/tem*.nc -g ombg -v sea_water_temperature -b -.2 .2 -c jet -q 0 -s $plotdir/temp_profile.png
+        ;;
+    esac
+    DH=$(($DH+$fctl))
+    YMDH=$(date -ud "$date_YMDH + $DH hours" +%Y%m%d%H )
+done  # day loop
+
+
+
+#mv *png $ddir/.
```
Comment (guillaumevernieres) [inline monitoring/godas_plotobs.sh]: remove comments

File: monitoring/soca_plotfield.py [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+        ax.set_ylim((-args.maxdepth, 0))
+        if  args.proj == 'hat10':
+            #ax.set_extent([-55,-10, 0, 50], ccrs.PlateCarree())
+            ax.set_xlim((-55,-15))
+        else:
+            ax.set_extent([-100, 0, 0, 50], ccrs.PlateCarree())
+        pngname=tail+'.'+args.variable+'.'+str(args.lat)+'.'+ str(args.maxdepth)+'.zonal.png'
+        plt.savefig(pngname)
+
+############################3333#!/usr/bin/env python3
```
Comment (guillaumevernieres) [inline monitoring/soca_plotfield.py]: remove comment

File: monitoring/soca_plotfield.sh [hunk 6]
```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+#Command:
+#   bash soca_plotfield.sh exp_name state_vector start_date num_of_days
+
+
```
Comment (guillaumevernieres) [inline monitoring/soca_plotfield.sh]: Same as other comments ... Go through that file and remove the debugging comments

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]

File: monitoring/godas_plotobs.py (hunk 2 above)
Comment (mjhossen) [reply monitoring/godas_plotobs.py]: resolved

File: monitoring/godas_plotobs.sh (hunk 3 above)
Comment (mjhossen) [reply monitoring/godas_plotobs.sh]: deleted

File: monitoring/godas_plotobs.sh (hunk 4 above)
Comment (mjhossen) [reply monitoring/godas_plotobs.sh]: removed

File: monitoring/soca_plotfield.py (hunk 5 above)
Comment (mjhossen) [reply monitoring/soca_plotfield.py]: removed

File: monitoring/soca_plotfield.py [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+                          transform=ccrs.PlateCarree(),
+                          cmap=colormap )
+    ax.gridlines()
+    plt.colorbar(p, ax=ax, shrink=0.5)
+    plt.title(varname)
+    if pretty == True:
+        ax.add_feature(cartopy.feature.LAND, edgecolor='black')
+        ax.add_feature(cartopy.feature.LAKES, edgecolor='black')
+        ax.coastlines()
+        #ax.background_img(name='BM', resolution='high')
```
Comment (ShastriPaturi) [inline monitoring/soca_plotfield.py]: remove comment

**Review by ShastriPaturi — CHANGES_REQUESTED** [review]
Please remove the comments, except for Usage,

**Comment by hyunchul386** [comment]
monitoring/Readmd.md, the variable name flag is "-v"

for example: bash soca_plotfield.sh -x EMC2 -v bkg -s 20210601 -e 20210601 bash soca_plotfield.sh -x EMC2 -v incr -s 20210601 -e 20210601

in the script of monitoring/godas_plotobs.sh, the variable name flag is "-t".  It needs to be fixed to -v or -t 

while getopts x:t:s:e: flag
do
    case "${flag}" in
        x) exp=${OPTARG};;  # experiment
        t) obs_type=${OPTARG};;  #variable name
        s) start_date=${OPTARG}Z00;;
        e) end_date=${OPTARG}00;;
    esac
done

File: monitoring/soca_plotfield.py (hunk 7 above)
Comment (mjhossen) [reply monitoring/soca_plotfield.py]: removed

**Comment by hyunchul386** [comment]
For me looks good. Adding captions in figure might be helpful

**Review by guillaumevernieres — APPROVED** [review]

File: monitoring/README.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+To plot the fields one can use the python scripts: soca_plotfield.py
+
+However, it is better to run a bash script to plot various variables on different dates.
+
+The command is :
+bash soca_plotfield.sh -x exp_name -v state_vector -s start_date -e end_date
```
Comment (ShastriPaturi) [inline monitoring/README.md]: Please provide instructions:
1. Where to run the soca_plotfield.sh from.
2. exp_name should have the path?
3. The output is expected to be in for e.g., /work/noaa/marine/Shastri.Paturi/PSL-final/monitoring/2019/

**Review by ShastriPaturi — CHANGES_REQUESTED** [review]

File: monitoring/soca_plotfield.sh [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+    case "${flag}" in
+        x) exp=${OPTARG};;  # experiment
+        v) stv=${OPTARG};;  #state vector
+        s) start_date=${OPTARG}Z00;;
+        e) end_date=${OPTARG}00;;
+    esac
+done
+echo experiments: $exp
+fctl=24
+cycle_dir=/work/noaa/marine/jhossen/$exp
```
Comment (ShastriPaturi) [inline monitoring/soca_plotfield.sh]: this should be generalized as $exp, since the user will specify exp path

File: monitoring/godas_plotobs.sh [hunk 10]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+#Command:
+#   bash godas_obsplot.sh -x exp_name -t obs_type -s start_date -e end_date
```
Comment (ShastriPaturi) [inline monitoring/godas_plotobs.sh]: The parameters in README.md and the one here are different, for eg.,
bash godas_obsplot.sh -x exp_name -t obs_type -s start_date -e end_date 
&
bash soca_plotfield.sh -x exp_name -v state_vector -s start_date -e end_date

**Review by ShastriPaturi — CHANGES_REQUESTED** [review]

File: monitoring/soca_plotfield.sh [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+        p) cycle_dir=${OPTARG};;  #state vector
+        s) start_date=${OPTARG}Z00;;
+        e) end_date=${OPTARG}00;;
+    esac
+done
+echo experiments: $exp
+fctl=24
+#cycle_dir=/work/noaa/marine/jhossen/$exp
+echo 'EXP dir:' ${cycle_dir}
+cdir=$PWD/$exp
```
Comment (ShastriPaturi) [inline monitoring/soca_plotfield.sh]: Remove $PWD. It is the directory where the scripts reside. Should be exppath

**Review by ShastriPaturi — CHANGES_REQUESTED** [review]

File: monitoring/godas_plotobs.sh (hunk 10 above)
Comment (mjhossen) [reply monitoring/godas_plotobs.sh]: fixed

File: monitoring/soca_plotfield.sh (hunk 11 above)
Comment (mjhossen) [reply monitoring/soca_plotfield.sh]: It should not be. $PWD is for the plotting dir.

File: monitoring/soca_plotfield.sh (hunk 9 above)
Comment (mjhossen) [reply monitoring/soca_plotfield.sh]: corrected with flag for exp path.

File: monitoring/README.md (hunk 8 above)
Comment (mjhossen) [reply monitoring/README.md]: resolved.

File: monitoring/soca_plotfield.sh [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+        x) exp=${OPTARG};;  # experiment
+        v) stv=${OPTARG};;  #state vector
+        p) cycle_dir=${OPTARG};;  #path/to/exp
+        s) start_date=${OPTARG}Z00;;
+        e) end_date=${OPTARG}00;;
+    esac
+done
+echo experiments: $exp
+fctl=24
+#cycle_dir=/work/noaa/marine/jhossen/$exp
```
Comment (ShastriPaturi) [inline monitoring/soca_plotfield.sh]: Add:
cycle_dir=$exp

**Review by ShastriPaturi — CHANGES_REQUESTED** [review]

**Comment by guillaumevernieres** [comment]
@ShastriPaturi & @hyunchul386 , can you finish/start your reviews, since you have actually tested and used @mjhossen 's scripts?

**Review by hyunchul386 — APPROVED** [review]

**Review by ShastriPaturi — APPROVED** [review]

**Comment by guillaumevernieres** [comment]
@mjhossen , you need to resolve the conflict before I merge this banch.

**Comment by mjhossen** [comment]
> @mjhossen , you need to resolve the conflict before I merge this banch.

Resolved!
